### PR TITLE
Add refresh token

### PR DESCRIPTION
**Pull request by TIove**
Relative https://github.com/LT-Students/DigitalOffice-UserService/pull/65

### CODE REVIEW

File: src/AuthService.Broker/Consumers/GetTokenConsumer.cs [hunk 1]
```diff
@@ -19,12 +20,15 @@ public async Task Consume(ConsumeContext<IGetTokenRequest> context)
         {
             var response = OperationResultWrapper.CreateResponse(GetTokenResult, context.Message);
 
-            await context.RespondAsync<IOperationResult<string>>(response);
+            await context.RespondAsync<IOperationResult<(string, string)>>(response);
         }
 
-        private string GetTokenResult(IGetTokenRequest request)
+        private (string, string) GetTokenResult(IGetTokenRequest request)
```
Comment (DarkFelas) [inline src/AuthService.Broker/Consumers/GetTokenConsumer.cs]: use named tuple
(string name1, string name2)

File: src/AuthService.Broker/Consumers/GetTokenConsumer.cs [hunk 2]
```diff
@@ -19,12 +20,15 @@ public async Task Consume(ConsumeContext<IGetTokenRequest> context)
         {
             var response = OperationResultWrapper.CreateResponse(GetTokenResult, context.Message);
 
-            await context.RespondAsync<IOperationResult<string>>(response);
+            await context.RespondAsync<IOperationResult<(string, string)>>(response);
```
Comment (DarkFelas) [inline src/AuthService.Broker/Consumers/GetTokenConsumer.cs]: named tuple

File: src/AuthService.Business/Commands/Interfaces/IRefreshTokenCommand.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+using LT.DigitalOffice.AuthService.Models.Dto.Requests;
+using LT.DigitalOffice.AuthService.Models.Dto.Responses;
+using LT.DigitalOffice.Kernel.Attributes;
+
+namespace LT.DigitalOffice.AuthService.Business.Commands.Interfaces
+{
+    [AutoInject]
+    public interface IRefreshTokenCommand
+    {
+        public LoginResult Execute(RefreshRequest request);
```
Comment (DarkFelas) [inline src/AuthService.Business/Commands/Interfaces/IRefreshTokenCommand.cs]: public inside interface?

File: src/AuthService.Business/Commands/LoginCommand.cs [hunk 4]
```diff
@@ -61,7 +62,8 @@ public async Task<LoginResult> Execute(LoginRequest request)
             var result = new LoginResult
             {
                 UserId = userCredentials.UserId,
-                Token = _tokenEngine.Create(userCredentials.UserId)
+                AccessToken = _tokenEngine.Create(userCredentials.UserId, TokenType.Access),
```
Comment (DarkFelas) [inline src/AuthService.Business/Commands/LoginCommand.cs:65]: do not forget to update spec and inform front team about new name

File: src/AuthService.Business/Commands/RefreshTokenCommand.cs [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+            var result = new LoginResult
+            {
+                UserId = userId,
+                AccessToken = _tokenEngine.Create(userId, TokenType.Access),
+                RefreshToken = _tokenEngine.Create(userId, TokenType.Refresh)
+            };
+            
+            _logger.LogInformation(
+                "User was successfully refreshed token with UserId: '{userId}' from IP: {requestIP}",
```
Comment (DarkFelas) [inline src/AuthService.Business/Commands/RefreshTokenCommand.cs]: User '{userId}' has successfully refreshed access token from IP '{userIP}'

File: src/AuthService.Models.Dto/Responses/LoginResult.cs [hunk 6]
```diff
@@ -5,6 +5,7 @@ namespace LT.DigitalOffice.AuthService.Models.Dto.Responses
     public record LoginResult
     {
         public Guid UserId { get; set; }
```
Comment (DarkFelas) [inline src/AuthService.Models.Dto/Responses/LoginResult.cs]: change to { get; init; }

File: src/AuthService.Token/Interfaces/ITokenEngine.cs [hunk 7]
```diff
@@ -7,6 +8,6 @@ public interface ITokenEngine
         /// <summary>
         /// Create new token based on user login.
```
Comment (DarkFelas) [inline src/AuthService.Token/Interfaces/ITokenEngine.cs]: update comment

File: src/AuthService.Token/TokenSettings.cs [hunk 8]
```diff
@@ -5,7 +5,8 @@
     /// </summary>
     public class TokenSettings
     {
-        public double TokenLifetimeInMinutes { get; set; }
+        public double AccessTokenLifetimeInMinutes { get; set; }
```
Comment (djino97) [inline src/AuthService.Token/TokenSettings.cs:8]: fix name -- AccessTokenLife**Time**InMinutes

**Review by djino97 — APPROVED** [review]

File: src/AuthService.Token/TokenSettings.cs (hunk 8 above)
Comment (TIove) [reply src/AuthService.Token/TokenSettings.cs:8]: That name already was there in settings. I thought it is not so bad, because lifetime is one word

**Review by BelehovEgor — APPROVED** [review]

**Review by MariyaSin — APPROVED** [review]

**Pull request merged**